cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1774,13 +1785,15 @@ def delete(self):
         """
         Delete this volume.
         """
-        (self._provider
-             .gce_compute
-             .disks()
-             .delete(project=self._provider.project_name,
-                     zone=self._provider.default_zone,
-                     disk=self.name)
-             .execute())
+        response = (self._provider
+                    .gce_compute
+                    .disks()
+                    .delete(project=self._provider.project_name,
+                            zone=self._provider.default_zone,
+                            disk=self.name)
+                    .execute())
+        self._provider.wait_for_operation(
+            response, zone=self._provider.default_zone)
```

Comment:
@baizhang Is there a specific reason to block till the delete is complete?? I think ideally, the delete call shouldn't block. I believe the tests handle this already, and wait for the delete to complete.

Reply:
This is the only issue that remains - should we just go ahead and merge it and we can sort this issue out later since it isn't a blocker?

Reply:
Thanks for your review, @nuwang !

I agree. Let's go ahead and merge this one first. And I will work on a non-blocking delete method in a separate pull request.